source/scripts/data/monsters.js

```diff
@@ -13,6 +13,34 @@ export default {
         health: 2,
         strength: 1,
     },
+    RED_BAT: {
+        sprite: DATA.SPRITES.MONSTERS.BAT,
+        color: DATA.COLORS.RED,
+        health: 1,
+        strength: 1,
+        movement: function (target, current) {
+            return [
+                {x: -1},
+                {x: +1},
+                {y: -1},
+                {y: +1}
+            ][Math.floor((Math.random() * 4))]
+        }
+    },
+    BLUE_BAT: {
+        sprite: DATA.SPRITES.MONSTERS.BAT,
+        color: DATA.COLORS.BLUE,
+        health: 1,
+        strength: 1,
+        movement: function (target, current) {
+            return [
+                {x: -1},
+                {x: +1},
+                {y: -1},
+                {y: +1}
+            ][Math.floor((Math.random() * 4))]
+        }
```

Comment:
So talk me through your reasoning here for passing the current position in as a parameter?

The `this` keyword in javascript is super weird, but one of it's strengths is when used in object oriented programming. When we call `this.movement = protomonster.movement || ...`, the `this` inside the movement function is now the monster instance.

So if the monster wants to know the current position in their movement function, they can just call `this.position`. Same goes for any other attributes the monster may have. And should the movement methods need to store some sort of state between each movement, they can store it by adding it to the monster instance via `this`.